### PR TITLE
feat: implement PXE frontend

### LINKED ISSUE #7: PXE Frontend

### PR DESCRIPTION
**Pull request by smira**
Also, improve error reporting and add more tests for invalid requests.

Fixes #7

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,7 +71,38 @@ Supported image paths:
 
 ## PXE Frontend API
 
-TBD
+PXE frontend provides an [iPXE script](https://github.com/mologie/talos-vmtoolsd) which automatically downloads and boots Talos.
```
Comment (frezbo) [inline README.md:74]: wrong url?

File: cmd/image-service/flags.go [hunk 2]
```diff
@@ -26,6 +26,8 @@ func initFlags() cmd.Options {
 
 	flag.IntVar(&opts.AssetBuildMaxConcurrency, "asset-builder-max-concurrency", cmd.DefaultOptions.AssetBuildMaxConcurrency, "maximum concurrency for asset builder")
 
+	flag.StringVar(&opts.SelfURL, "self-url", cmd.DefaultOptions.SelfURL, "service external endpoint URL")
```
Comment (frezbo) [inline cmd/image-service/flags.go:29]:
```suggestion
	flag.StringVar(&opts.SelfURL, "external-url", cmd.DefaultOptions.SelfURL, "service external endpoint URL")
```

maybe?

File: README.md (hunk 1 above)
Comment (smira) [reply README.md:74]: yes :facepalm:

File: cmd/image-service/flags.go (hunk 2 above)
Comment (smira) [reply cmd/image-service/flags.go:29]: yes, good point

**Review by frezbo — APPROVED** [review]

**Pull request merged**